### PR TITLE
Add google-perftools as 32bit

### LINKED ISSUE #69: "Half-Life 2: Update" does not launch because of missing shared library
"Half-Life 2: Update", a linux-compatible-game, cannot launch. Here is the relevant error message from the terminal:

~/.var/app/com.valvesoftware.Steam/data/Steam/steamapps/common/Half-Life 2 Update/hl2_linux: error while loading shared libraries: libtcmalloc_minimal.so.4: cannot open shared object file: No such file or directory

### PR DESCRIPTION
**Pull request by nanonyme**
Fixes #69 #105

### CODE REVIEW

File: modules/compat-libraries-32bit.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+x-compat-i386-opts: &compat_i386_opts
+  prepend-pkg-config-path: /app/lib32/pkgconfig:/usr/lib/i386-linux-gnu/pkgconfig
+  ldflags: -L/app/lib32
+  prepend-path: /usr/lib/sdk/toolchain-i386/bin
+  env:
+    CC: i686-unknown-linux-gnu-gcc
+    CXX: i686-unknown-linu-gnu-gcc
```
Comment (gasinvein) [inline modules/compat-libraries-32bit.yml]:
```suggestion
    CXX: i686-unknown-linux-gnu-g++
```

File: modules/compat-libraries-32bit.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+modules:
+  - name: google-perfools
+    buildsystem: autotools
+    build-options:
+      arch:
+        x86_64: *compat_i386_opts
+    sources:
+      - type: git
+        url: https://github.com/gperftools/gperftools.git
+        commit: perftools-1.6-0-gfd52267cb91a42f839c7be6a348e014cfdbcaca9
```
Comment (gasinvein) [inline modules/compat-libraries-32bit.yml]: Seems like gperftools 1.6 won't build with newer glibc: gperftools/gperftools#350
```suggestion
        commit: 0b87a55895a184bcc529b4ce13c46c0411182b85
        tag: google-perftools-1.8.3
```

Comment (nanonyme) [reply modules/compat-libraries-32bit.yml]: Is this just head or some specific commit?

Comment (gasinvein) [reply modules/compat-libraries-32bit.yml]: It's the commit tagged as version 1.8.3.

**Comment by gasinvein** [comment]
@nanonyme Maybe rename this module to something more meaningful, e.g. `game-libraries-32bit.yml`? Other module names clearly tell what they are needed for, but not this one.

File: modules/compat-libraries-32bit.yml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+x-compat-i386-opts: &compat_i386_opts
+  prepend-pkg-config-path: /app/lib32/pkgconfig:/usr/lib/i386-linux-gnu/pkgconfig
+  ldflags: -L/app/lib32
+  prepend-path: /usr/lib/sdk/toolchain-i386/bin
+  env:
+    CC: i686-unknown-linux-gnu-gcc
+    CXX: i686-unknown-linux-gnu-g++
+    libdir: /app/lib32
```
Comment (gasinvein) [inline modules/compat-libraries-32bit.yml]: `libdir` here is supposedly not an env var, but a build options, which tells the build system where to install libraries, similarly to `prefix`.
```suggestion
  libdir: /app/lib32
```

**Comment by nanonyme** [comment]
I'm going to leave renaming for after this since these further work depending on this MR.

**Comment by gasinvein** [comment]
@nanonyme  Are you planning to put _all_ 32bit libraries into the `compat-libraries-32bit.yml` module?

**Comment by nanonyme** [comment]
@gasinvein sadly probably yes. Otherwise there will be *massive* repetition of the 32bit definitions since cross-include anchors aren't a thing.

**Comment by gasinvein** [comment]
@nanonyme I believe repeating 32bit definitions is less evil than repeating sources and config-opts. The later could cause inconsistency between 64 and 32-bit builds of the same library.

**Pull request merged**